File: day_2/day_2.py
import os

__location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
with open(os.path.join(__location__,'input.txt')) as file:
    commands = file.readlines()
    commands = [command.rstrip().split(' ') for command in commands]
    

horizontal = 0
depth = 0

# # Part 1
for command_map in commands:
    if command_map[0] == 'forward':
        horizontal = horizontal + int(command_map[1])
    elif command_map[0] == 'up':
        depth = depth - int(command_map[1])
    elif command_map[0] == 'down':
        depth = depth + int(command_map[1])
print(horizontal * depth)

# Part 2
aim = 0

for command_map in commands:
    if command_map[0] == 'down':
        aim = aim + int(command_map[1])
    elif command_map[0] == 'up':
        aim = aim - int(command_map[1])
    elif command_map[0] == 'forward':
        horizontal = horizontal + int(command_map[1])
        depth = depth + (int(command_map[1]) * aim)
print(horizontal * depth)

File: day_2/day_2.dart
// https://adventofcode.com/2021/day/2

import 'dart:io';

main() async {
  final File file = File(Directory.current.path + '/day_2/input.txt');
  List<String> values = await file.readAsLines();

  List<List> commands = [];
  values.forEach((value) {
    commands.add(value.split(' '));
  });

  //! PART 1
  int horizontal = 0;
  int depth = 0;

  commands.forEach((command) {
    int unit = int.parse(command[1]);

    switch (command[0]) {
      case 'forward':
        horizontal += unit;
        break;
      case 'up':
        depth -= unit;
        break;
      case 'down':
        depth += unit;
    }
  });

  print(horizontal * depth);

  //! PART 2
  horizontal = 0;
  depth = 0;
  int aim = 0;

  commands.forEach((command) {
    int unit = int.parse(command[1]);

    switch (command[0]) {
      case 'forward':
        horizontal += unit;
        depth += (unit * aim);
        break;
      case 'up':
        aim -= unit;
        break;
      case 'down':
        aim += unit;
    }
  });

  print(horizontal * depth);
}
